tests/js/comments/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -3614,9 +3614,9 @@ function binary() {
 
 function binaryInBinaryLeft() {
   return (
-    // Reason for 42
+    (// Reason for 42
```

Comment:
Ugly

Reply:
Ooh yeah good catch, @evilebottnawi -- seems like the comment is being considered part of the parent operation. I'll see if I can fix this. If you happen to have any advice on where to look i'd appreciate it!

Reply:
@evilebottnawi this is my first time contributing here and I'm trying to dig into this a bit. It doesn't seem like from the context of needs-parens that you can really specify where the parens should be added, or add a linebreak after a comment. However, running prettier a second time against the input yields the much cleaner:

```js
function binaryInBinaryLeft() {
  return (
    // Reason for 42
    (42 * 84) +
    2
  );
}
```

So I guess there are two options one of which I'm pretty unsure of where to look for examples of how to achieve the effect:

#### Option 1: Skip Nodes with comments
Essentially we'd just look if either the left or right side of the node has comments and return false in needs-parens. This would allow us to care for the vast majority of `mixed-operators` issues. Remaining issues that end users are having can be resolved by moving the comments manually.

#### Option 2: Change the starting line/col of parenthesis

The problem with the output stems from the comment being a part of the expression. I tried poking around in printer-estree with `parts` which is having parenthesis injected into it if needsParens is `true`. However, I'm not sure what the desired output should be or how to achieve it.

```js
function binaryInBinaryLeft() {
  return (
    // Reason for 42
    (42 * 84) + 2
  );
}
```

seems like the right approach, but I'm getting a little lost on the proper way to achieve this result. If anyone reading this has pointers I'm all ears and very eager to have this PR be considered for inclusion in prettier.

Reply:
I went with the first option for now.